python/ray/workflow/workflow_storage.py

```diff
@@ -621,6 +677,21 @@ def delete_workflow(self) -> None:
         if not found:
             raise WorkflowNotFoundError(self._workflow_id)
 
+    def update_workflow_status(
+        self, status: WorkflowStatus, prev_status: Optional[WorkflowStatus] = None
+    ):
```

Comment:
It seems a little bit complicated to me. Not quite sure why do we need `prev_status`. We can sync offline about this.